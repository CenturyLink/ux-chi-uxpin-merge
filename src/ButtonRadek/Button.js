import React from 'react';
import PropTypes from 'prop-types';
import ButtonStyles from './Button.styles';

const Button = (props) => (
  <ButtonStyles {...props}>
    {props.icon}
    {props.children}
  </ButtonStyles>
);

Button.propTypes = {
  onClick: PropTypes.func,
  disabled: PropTypes.bool,
  type: PropTypes.oneOf(['primary', 'secondary', 'success', 'error', 'warning']),
  mode: PropTypes.oneOf(['filled', 'ghost', 'minimal', 'flat']),
  title: PropTypes.string,
  background: PropTypes.string,
  size: PropTypes.oneOf(['xs', 's', 'm', 'l', 'xl', 'xxl', 'xxxl']),
  stretched: PropTypes.bool,
  children: PropTypes.string,
  icon: PropTypes.node,
  iconDirection: PropTypes.oneOf(['left', 'right']),
};

Button.defaultProps = {
  iconDirection: 'left',
  disabled: false,
  mode: 'filled',
  size: 'm',
  stretched: true,
  type: 'primary',
};

export { Button as default };
