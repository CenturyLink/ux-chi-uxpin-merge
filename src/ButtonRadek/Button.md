# Button

```javascript
<Button
    stretched={true}
    type="error"
    mode="filled"
    icon={<Icon size="s" icon="TickerSvg" />}
    size="s"
    >Merge!</Button>
```