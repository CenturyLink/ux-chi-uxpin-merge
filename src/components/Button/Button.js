import * as PropTypes from 'prop-types';
import * as React from 'react';
import ButtonStyles from './Button.styles';

const Button = (props) => (
  <ButtonStyles {...props}>
    {props.icon}
    {props.children}
  </ButtonStyles>
);

/* eslint-disable sort-keys */
Button.propTypes = {
  onClick: PropTypes.func,
  disabled: PropTypes.bool,
  type: PropTypes.oneOf(['primary', 'secondary', 'success', 'error', 'warning']),
  mode: PropTypes.oneOf(['filled', 'ghost', 'minimal', 'flat']),
  /** @uxpinignoreprop */
  title: PropTypes.string,
  /** @uxpinignoreprop */
  background: PropTypes.string,
  /**
   * @uxpinpropname Label
   */
  children: PropTypes.string,
  icon: PropTypes.node,
  iconDirection: PropTypes.oneOf(['left', 'right']),
  size: PropTypes.oneOf(['xs', 's', 'm', 'l', 'xl', 'xxl', 'xxxl']),
  stretched: PropTypes.bool,
};
/* eslint-enable sort-keys */

Button.defaultProps = {
  disabled: false,
  iconDirection: 'left',
  mode: 'filled',
  size: 'm',
  stretched: true,
  type: 'primary',
};

export { Button as default };
