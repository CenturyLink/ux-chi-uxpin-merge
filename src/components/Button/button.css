.chi button {
  transform: scale(1);
  transition-duration: 0.1s;
}

.chi button.-hit {
  transform: scale(1.1);
}
