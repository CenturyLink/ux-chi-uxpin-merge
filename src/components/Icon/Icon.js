import PropTypes from 'prop-types';
import React from 'react';
import sizes from '../../styles/sizes';
import * as Icons from './icons';

const getIconSize = (size) => {
  switch (size) {
    case 'xs':
    case 's':
    case 'm':
    case 'l':
    case 'xl':
    case 'xxl':
    case 'xxxl':
      return `${sizes[size]}px`;

    default:
      return `${sizes.xs}px`;
  }
};

export default function Icon({ color, icon, size }) {
  const iconSize = getIconSize(size);

  return React.createElement(Icons[icon], {
    fill: color,
    height: iconSize,
    width: iconSize,
  });
}

Icon.propTypes = {
  color: PropTypes.string,
  icon: PropTypes.oneOf([
    'ActionAddToFavoritesSvg',
    'ActionArchiveSvg',
    'ActionArrangeLeftSvg',
    'ActionArrangeRightSvg',
    'ActionArrangeVerticalCenterSvg',
    'ActionBreakSymbolSvg',
    'ActionChangeProjectGroupSvg',
    'ActionCloneSvg',
    'ActionCloseSvg',
    'ActionDetachingHandlerSvg',
    'ActionDisableSvg',
    'ActionDistributeVerticallySvg',
    'ActionDuplicateSvg',
    'ActionElementAlignCenterSvg',
    'ActionElementAlignSvg',
    'ActionElementFlipSvg',
    'ActionEnableSvg',
    'ActionExitSvg',
    'ActionFullScreenSvg',
    'ActionInsertDataSvg',
    'ActionLogOutSvg',
    'ActionMoveBehindSvg',
    'ActionMoveBySvg',
    'ActionMoveForwardSvg',
    'ActionMoveToBackSvg',
    'ActionMoveToFrontSvg',
    'ActionMoveToSvg',
    'ActionMultipropertyLockSvg',
    'ActionMultipropertyUnlockBordersSvg',
    'ActionMultipropertyUnlockCornersSvg',
    'ActionNextStateSvg',
    'ActionPathBreakSvg',
    'ActionPathDefaultSvg',
    'ActionPathDifferenceSvg',
    'ActionPathIntersectSvg',
    'ActionPathSubtractSvg',
    'ActionPathUnionSvg',
    'ActionPreviousStateSvg',
    'ActionRemoveFromGroupSvg',
    'ActionSetContentSvg',
    'ActionSwitchSvg',
    'ActionSymbolSyncSvg',
    'ActionUnarchiveSvg',
    'AdjustSettingsSvg',
    'AdvancedAnimationSvg',
    'AngleSvg',
    'ArrowAxisSvg',
    'ArrowEnterSvg',
    'ArrowGoBackSvg',
    'ArrowPlaySvg',
    'ArrowRedoSvg',
    'ArrowRepeatSvg',
    'ArrowReplaceSvg',
    'ArrowRetrieveSvg',
    'ArrowRotateSvg',
    'ArrowShowSvg',
    'ArrowSizeSvg',
    'ArrowThinSvg',
    'ArrowUndoSvg',
    'ArrowSvg',
    'BrandAppleSvg',
    'BrandGoogleSimpleSvg',
    'BrandGoogleSvg',
    'BrandSketchSvg',
    'BrandSlackSvg',
    'BrandTwitterSvg',
    'BrandTypekitSvg',
    'BrandUxpinSvg',
    'BrandWindowsSvg',
    'BuildingSvg',
    'CalendarSvg',
    'ChangelogSvg',
    'ChevronSvg',
    'ClockSvg',
    'CodeBlockSvg',
    'CodeInlineSvg',
    'ColorDropSvg',
    'ColorsDropperSvg',
    'ColorsSchemesSvg',
    'ColorsSwatchesSvg',
    'ConnectionConnectedSvg',
    'ConnectionDisconnectedSvg',
    'CursorCrosshairSvg',
    'CursorDefaultSvg',
    'CursorPointerSvg',
    'DesktopAppSvg',
    'DotsSvg',
    'ElementsBoxSvg',
    'ElementsButtonSvg',
    'ElementsCheckboxSvg',
    'ElementsCircleSvg',
    'ElementsCodeComponentSvg',
    'ElementsElementSvg',
    'ElementsGenericSvg',
    'ElementsGroupingSvg',
    'ElementsHotspotSvg',
    'ElementsIconSvg',
    'ElementsImageSvg',
    'ElementsInputSvg',
    'ElementsMultiSelectSvg',
    'ElementsMultistatesEditSvg',
    'ElementsMultistatesSvg',
    'ElementsPathSvg',
    'ElementsRadioSvg',
    'ElementsSelectListSvg',
    'ElementsSliceSvg',
    'ElementsSymbolSvg',
    'ElementsTableSvg',
    'ElementsTextBlockSvg',
    'ElementsTextElementSvg',
    'ElementsTextareaSvg',
    'ElementsUngroupingSvg',
    'ElementsVideoSvg',
    'EndpointEmptyCircleSvg',
    'EndpointEmptyRectangleSvg',
    'EndpointEmptyRhombSvg',
    'EndpointEmptyTriangleSvg',
    'EndpointFilledArrowSvg',
    'EndpointFilledCircleSvg',
    'EndpointFilledRectangleSvg',
    'EndpointFilledRhombSvg',
    'EndpointFilledTriangleSvg',
    'EndpointLineSvg',
    'EndpointMirroredTriangleSvg',
    'EndpointNoneSvg',
    'EndpointOpenedTriangleSvg',
    'ExternalLinkSvg',
    'FeatureAnimationsSvg',
    'FeatureApprovalSvg',
    'FeatureChatSvg',
    'FeatureCommentsSvg',
    'FeatureDocumentationSvg',
    'FeatureGridSvg',
    'FeatureIterationSvg',
    'FeatureLayersSvg',
    'FeatureLivePresentationSvg',
    'FeatureMobileSvg',
    'FeatureResponsiveSvg',
    'FeatureSitemapSvg',
    'FeatureUsabilityTestingSvg',
    'FilesDownloadCloudSvg',
    'FilesUploadSvg',
    'FolderClosedOutlineSvg',
    'FolderClosedSvg',
    'FolderOpenedSvg',
    'GearSvg',
    'HeartSvg',
    'InfoSimpleSvg',
    'InfoWarningSvg',
    'LaptopSvg',
    'LayoutListSvg',
    'LayoutMediumTileSvg',
    'LayoutSmallTileSvg',
    'LibrariesDesignLibrariesSvg',
    'LibrariesMoreSvg',
    'LibrariesSimpleSvg',
    'LinkChainBreakSvg',
    'LinkChainSvg',
    'ListIndentationAddSvg',
    'ListIndentationRemoveSvg',
    'ListOrderedSvg',
    'ListUnorderedSvg',
    'MailSvg',
    'MaskSvg',
    'MinusOutlineSvg',
    'OpacitySvg',
    'OptionsViewSvg',
    'PadlockBrokenSvg',
    'PadlockLockedSvg',
    'PadlockUnlockedSvg',
    'PenToolSvg',
    'PencilSvg',
    'PiggyBankSvg',
    'PinMapSolidSvg',
    'PinMapSvg',
    'PinSimpleSvg',
    'PluginsSvg',
    'PlusOutlineSvg',
    'PlusSvg',
    'QuestionMarkSolidSvg',
    'QuestionMarkSvg',
    'ScrollToSvg',
    'SearchSvg',
    'ShapeDiamondSvg',
    'ShapeRhombusSvg',
    'ShapeStarSvg',
    'ShapeTriangleRightAngledSvg',
    'ShapeTriangleSvg',
    'ShareClosedSvg',
    'ShareOpenedSvg',
    'ShoppingCartSvg',
    'SoundMuteSvg',
    'SoundUnmuteSvg',
    'StateBaseSvg',
    'StatusApprovedSvg',
    'StatusPausedSvg',
    'StatusProgressSvg',
    'StatusRejectedSvg',
    'StatusWaitingSvg',
    'StrokeEndButtSvg',
    'StrokeEndRoundSvg',
    'StrokeEndSquareSvg',
    'StrokeJoinBevelSvg',
    'StrokeJoinMiterSvg',
    'StrokeJoinRoundSvg',
    'StrokePointAsymmetricSvg',
    'StrokePointDisconnectedSvg',
    'StrokePointMirroredSvg',
    'StrokePointStraightSvg',
    'ThunderSvg',
    'TickerSvg',
    'ToggleSvg',
    'TooltipLeftSvg',
    'TooltipTopSvg',
    'TrashSvg',
    'TypographyAlignCenterSvg',
    'TypographyAlignJustifySvg',
    'TypographyAlignLeftSvg',
    'TypographyCustomFontSvg',
    'TypographyFontSizeSvg',
    'TypographyLetterSpacingSvg',
    'TypographyLineHeightSvg',
    'UserAddSvg',
    'UserSimpleSvg',
    'VariableSvg',
    'VisibilityOutlineHiddenSvg',
    'VisibilityOutlineVisibleSvg',
    'VisibilitySolidHiddenSvg',
    'VisibilitySolidVisibleSvg',
  ]),
  size: PropTypes.oneOf(['xxs', 'xs', 's', 'm', 'l', 'xl', 'xxl', 'xxxl']),
};
