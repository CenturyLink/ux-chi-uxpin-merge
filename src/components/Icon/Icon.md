# Icon

```javascript
<Icon size="l" icon="PinMapSolidSvg"/>
```