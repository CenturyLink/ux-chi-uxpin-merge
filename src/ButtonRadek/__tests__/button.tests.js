import React from 'react';
import * as enzyme from 'enzyme';
import toJson from 'enzyme-to-json';
import Button from '../Button';
import Icon from '../../Icon/Icon';


test('renders error Button correctly', () => {
  const button = (
    <Button
      stretched={false}
      type="error"
      mode="filled">
Test
    </Button>
  );

  expect(toJson(enzyme.mount(button))).toMatchSnapshot('enzyme.mount');
});

test('renders success Button correctly', () => {
  const button = (
    <Button
      stretched={false}
      type="success"
      mode="filled">
Test
    </Button>
  );

  expect(toJson(enzyme.mount(button))).toMatchSnapshot('enzyme.mount');
});


test('renders warning Button correctly', () => {
  const button = (
    <Button
      stretched={false}
      type="warning"
      mode="filled">
Test
    </Button>
  );

  expect(toJson(enzyme.mount(button))).toMatchSnapshot('enzyme.mount');
});

test('renders primary Button correctly', () => {
  const button = (
    <Button
      stretched={false}
      type="primary"
      mode="filled">
Test
    </Button>
  );

  expect(toJson(enzyme.mount(button))).toMatchSnapshot('enzyme.mount');
});

test('renders secondary Button correctly', () => {
  const button = (
    <Button
      stretched={false}
      type="secondary"
      mode="filled">
Test
    </Button>
  );

  expect(toJson(enzyme.mount(button))).toMatchSnapshot('enzyme.mount');
});

test('renders primary ghost Button correctly', () => {
  const button = (
    <Button
      stretched={false}
      type="primary"
      mode="ghost">
Test
    </Button>
  );

  expect(toJson(enzyme.mount(button))).toMatchSnapshot('enzyme.mount');
});

test('renders icon Button correctly', () => {
  const button = (
    <Button
      stretched={false}
      type="error"
      icon={<Icon size="s" icon="TickerSvg" />}
      iconDirection="left"
      mode="filled"
    />
  );

  expect(toJson(enzyme.shallow(button))).toMatchSnapshot('enzyme.shallow');
});

test('renders Button with icon correctly', () => {
  const button = (
    <Button
      stretched={false}
      type="error"
      icon={<Icon size="s" icon="TickerSvg" />}
      iconDirection="left"
      mode="filled">
Test
    </Button>
  );

  expect(toJson(enzyme.shallow(button))).toMatchSnapshot('enzyme.shallow');
});
