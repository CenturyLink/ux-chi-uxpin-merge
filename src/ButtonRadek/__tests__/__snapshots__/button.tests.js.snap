// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Button with icon correctly: enzyme.shallow 1`] = `
<Styled(withProps(Styled(button)))
  disabled={false}
  icon={
    <Icon
      icon="TickerSvg"
      size="s"
    />
  }
  iconDirection="left"
  mode="filled"
  size="m"
  stretched={false}
  type="error"
>
  <Icon
    icon="TickerSvg"
    size="s"
  />
  Test
</Styled(withProps(Styled(button)))>
`;

exports[`renders error Button correctly: enzyme.mount 1`] = `
.emotion-3 {
  background-color: #e20c00;
  color: white;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: auto;
  padding: 0 16px 0 16px;
  border-radius: 3px;
  border: 1px solid #e20c00;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen-Sans,Ubuntu,Cantarell,'Helvetica Neue',sans-serif;
  font-weight: 600;
  line-height: 31.200000000000003px;
  font-size: 12px;
  background-color: #e20c00;
  color: white;
}

.emotion-1:hover {
  background-color: #fb0d00;
  border: 1px solid #fb0d00;
  box-shadow: 4px 4px 15px rgba(200,11,0,0.1);
}

.emotion-1:focus {
  outline: none;
  background-color: #c80b00;
  border: 1px solid #950800;
}

.emotion-1:active {
  background-color: #c80b00;
  border: 1px solid #c80b00;
  box-shadow: 'none';
}

.emotion-1:disabled {
  background-color: #f6f6f6;
  border: 1px solid #f3f3f3;
  color: #cdcdcd;
}

.emotion-1:disabled:hover {
  box-shadow: none;
}

.emotion-1 svg {
  margin: 0 8px 0 0;
  fill: white;
}

<Button
  disabled={false}
  iconDirection="left"
  mode="filled"
  size="m"
  stretched={false}
  type="error"
>
  <Styled(withProps(Styled(button)))
    disabled={false}
    iconDirection="left"
    mode="filled"
    size="m"
    stretched={false}
    type="error"
  >
    <withProps(Styled(button))
      className="emotion-3 emotion-0"
      disabled={false}
      iconDirection="left"
      mode="filled"
      size="m"
      stretched={false}
      type="error"
    >
      <Styled(button)
        className="emotion-3 emotion-0"
        createProps={[Function]}
        disabled={false}
        iconDirection="left"
        mode="filled"
        size="m"
        stretched={false}
        type="error"
      >
        <button
          className="emotion-0 emotion-1 emotion-2"
          disabled={false}
          mode="filled"
          size="m"
          type="error"
        >
          Test
        </button>
      </Styled(button)>
    </withProps(Styled(button))>
  </Styled(withProps(Styled(button)))>
</Button>
`;

exports[`renders icon Button correctly: enzyme.shallow 1`] = `
<Styled(withProps(Styled(button)))
  disabled={false}
  icon={
    <Icon
      icon="TickerSvg"
      size="s"
    />
  }
  iconDirection="left"
  mode="filled"
  size="m"
  stretched={false}
  type="error"
>
  <Icon
    icon="TickerSvg"
    size="s"
  />
</Styled(withProps(Styled(button)))>
`;

exports[`renders primary Button correctly: enzyme.mount 1`] = `
.emotion-3 {
  background-color: #006cff;
  color: white;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: auto;
  padding: 0 16px 0 16px;
  border-radius: 3px;
  border: 1px solid #006cff;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen-Sans,Ubuntu,Cantarell,'Helvetica Neue',sans-serif;
  font-weight: 600;
  line-height: 31.200000000000003px;
  font-size: 12px;
  background-color: #006cff;
  color: white;
}

.emotion-1:hover {
  background-color: #1a7bff;
  border: 1px solid #1a7bff;
  box-shadow: 4px 4px 15px rgba(179,211,255,0.1);
}

.emotion-1:focus {
  outline: none;
  background-color: #0056cc;
  border: 1px solid 004cb3;
}

.emotion-1:active {
  background-color: #0056cc;
  border: 1px solid #0056cc;
  box-shadow: 'none';
}

.emotion-1:disabled {
  background-color: #f6f6f6;
  border: 1px solid #f3f3f3;
  color: #cdcdcd;
}

.emotion-1:disabled:hover {
  box-shadow: none;
}

.emotion-1 svg {
  margin: 0 8px 0 0;
  fill: white;
}

<Button
  disabled={false}
  iconDirection="left"
  mode="filled"
  size="m"
  stretched={false}
  type="primary"
>
  <Styled(withProps(Styled(button)))
    disabled={false}
    iconDirection="left"
    mode="filled"
    size="m"
    stretched={false}
    type="primary"
  >
    <withProps(Styled(button))
      className="emotion-3 emotion-0"
      disabled={false}
      iconDirection="left"
      mode="filled"
      size="m"
      stretched={false}
      type="primary"
    >
      <Styled(button)
        className="emotion-3 emotion-0"
        createProps={[Function]}
        disabled={false}
        iconDirection="left"
        mode="filled"
        size="m"
        stretched={false}
        type="primary"
      >
        <button
          className="emotion-0 emotion-1 emotion-2"
          disabled={false}
          mode="filled"
          size="m"
          type="primary"
        >
          Test
        </button>
      </Styled(button)>
    </withProps(Styled(button))>
  </Styled(withProps(Styled(button)))>
</Button>
`;

exports[`renders primary ghost Button correctly: enzyme.mount 1`] = `
.emotion-3 {
  background-color: none;
  color: #006cff;
}

.emotion-3 svg {
  fill: #006cff;
}

.emotion-3:hover,
.emotion-3:active,
.emotion-3:focus {
  color: white;
}

.emotion-3:hover svg,
.emotion-3:active svg,
.emotion-3:focus svg {
  fill: white;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: auto;
  padding: 0 16px 0 16px;
  border-radius: 3px;
  border: 1px solid #006cff;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen-Sans,Ubuntu,Cantarell,'Helvetica Neue',sans-serif;
  font-weight: 600;
  line-height: 31.200000000000003px;
  font-size: 12px;
  background-color: none;
  color: #006cff;
}

.emotion-1:hover {
  background-color: #1a7bff;
  border: 1px solid #1a7bff;
  box-shadow: 4px 4px 15px rgba(179,211,255,0.1);
}

.emotion-1:focus {
  outline: none;
  background-color: #0056cc;
  border: 1px solid 004cb3;
}

.emotion-1:active {
  background-color: #0056cc;
  border: 1px solid #0056cc;
  box-shadow: 'none';
}

.emotion-1:disabled {
  background-color: #f6f6f6;
  border: 1px solid #f3f3f3;
  color: #cdcdcd;
}

.emotion-1:disabled:hover {
  box-shadow: none;
}

.emotion-1 svg {
  margin: 0 8px 0 0;
  fill: white;
}

.emotion-1 svg {
  fill: #006cff;
}

.emotion-1:hover,
.emotion-1:active,
.emotion-1:focus {
  color: white;
}

.emotion-1:hover svg,
.emotion-1:active svg,
.emotion-1:focus svg {
  fill: white;
}

<Button
  disabled={false}
  iconDirection="left"
  mode="ghost"
  size="m"
  stretched={false}
  type="primary"
>
  <Styled(withProps(Styled(button)))
    disabled={false}
    iconDirection="left"
    mode="ghost"
    size="m"
    stretched={false}
    type="primary"
  >
    <withProps(Styled(button))
      className="emotion-3 emotion-0"
      disabled={false}
      iconDirection="left"
      mode="ghost"
      size="m"
      stretched={false}
      type="primary"
    >
      <Styled(button)
        className="emotion-3 emotion-0"
        createProps={[Function]}
        disabled={false}
        iconDirection="left"
        mode="ghost"
        size="m"
        stretched={false}
        type="primary"
      >
        <button
          className="emotion-0 emotion-1 emotion-2"
          disabled={false}
          mode="ghost"
          size="m"
          type="primary"
        >
          Test
        </button>
      </Styled(button)>
    </withProps(Styled(button))>
  </Styled(withProps(Styled(button)))>
</Button>
`;

exports[`renders secondary Button correctly: enzyme.mount 1`] = `
.emotion-3 {
  background-color: #f3f3f3;
  color: #444;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: auto;
  padding: 0 16px 0 16px;
  border-radius: 3px;
  border: 1px solid #f3f3f3;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen-Sans,Ubuntu,Cantarell,'Helvetica Neue',sans-serif;
  font-weight: 600;
  line-height: 31.200000000000003px;
  font-size: 12px;
  background-color: #f3f3f3;
  color: #444;
}

.emotion-1:hover {
  background-color: #f6f6f6;
  border: 1px solid #f6f6f6;
  box-shadow: 4px 2px 35px rgba(243,243,243,0.6);
}

.emotion-1:focus {
  outline: none;
  background-color: #f0f0f0;
  border: 1px solid 004cb3;
}

.emotion-1:active {
  background-color: #f0f0f0;
  border: 1px solid #f0f0f0;
  box-shadow: 'none';
}

.emotion-1:disabled {
  background-color: #f6f6f6;
  border: 1px solid #f3f3f3;
  color: #cdcdcd;
}

.emotion-1:disabled:hover {
  box-shadow: none;
}

.emotion-1 svg {
  margin: 0 8px 0 0;
  fill: white;
}

<Button
  disabled={false}
  iconDirection="left"
  mode="filled"
  size="m"
  stretched={false}
  type="secondary"
>
  <Styled(withProps(Styled(button)))
    disabled={false}
    iconDirection="left"
    mode="filled"
    size="m"
    stretched={false}
    type="secondary"
  >
    <withProps(Styled(button))
      className="emotion-3 emotion-0"
      disabled={false}
      iconDirection="left"
      mode="filled"
      size="m"
      stretched={false}
      type="secondary"
    >
      <Styled(button)
        className="emotion-3 emotion-0"
        createProps={[Function]}
        disabled={false}
        iconDirection="left"
        mode="filled"
        size="m"
        stretched={false}
        type="secondary"
      >
        <button
          className="emotion-0 emotion-1 emotion-2"
          disabled={false}
          mode="filled"
          size="m"
          type="secondary"
        >
          Test
        </button>
      </Styled(button)>
    </withProps(Styled(button))>
  </Styled(withProps(Styled(button)))>
</Button>
`;

exports[`renders success Button correctly: enzyme.mount 1`] = `
.emotion-3 {
  background-color: #63ad0e;
  color: white;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: auto;
  padding: 0 16px 0 16px;
  border-radius: 3px;
  border: 1px solid #63ad0e;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen-Sans,Ubuntu,Cantarell,'Helvetica Neue',sans-serif;
  font-weight: 600;
  line-height: 31.200000000000003px;
  font-size: 12px;
  background-color: #63ad0e;
  color: white;
}

.emotion-1:hover {
  background-color: #70c510;
  border: 1px solid #70c510;
  box-shadow: 4px 4px 15px rgba(99,173,14,0.1);
}

.emotion-1:focus {
  outline: none;
  background-color: #55950c;
  border: 1px solid #3a6608;
}

.emotion-1:active {
  background-color: #55950c;
  border: 1px solid #55950c;
  box-shadow: 'none';
}

.emotion-1:disabled {
  background-color: #f6f6f6;
  border: 1px solid #f3f3f3;
  color: #cdcdcd;
}

.emotion-1:disabled:hover {
  box-shadow: none;
}

.emotion-1 svg {
  margin: 0 8px 0 0;
  fill: white;
}

<Button
  disabled={false}
  iconDirection="left"
  mode="filled"
  size="m"
  stretched={false}
  type="success"
>
  <Styled(withProps(Styled(button)))
    disabled={false}
    iconDirection="left"
    mode="filled"
    size="m"
    stretched={false}
    type="success"
  >
    <withProps(Styled(button))
      className="emotion-3 emotion-0"
      disabled={false}
      iconDirection="left"
      mode="filled"
      size="m"
      stretched={false}
      type="success"
    >
      <Styled(button)
        className="emotion-3 emotion-0"
        createProps={[Function]}
        disabled={false}
        iconDirection="left"
        mode="filled"
        size="m"
        stretched={false}
        type="success"
      >
        <button
          className="emotion-0 emotion-1 emotion-2"
          disabled={false}
          mode="filled"
          size="m"
          type="success"
        >
          Test
        </button>
      </Styled(button)>
    </withProps(Styled(button))>
  </Styled(withProps(Styled(button)))>
</Button>
`;

exports[`renders warning Button correctly: enzyme.mount 1`] = `
.emotion-3 {
  background-color: #ff7d3a;
  color: white;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: auto;
  padding: 0 16px 0 16px;
  border-radius: 3px;
  border: 1px solid #ff7d3a;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen-Sans,Ubuntu,Cantarell,'Helvetica Neue',sans-serif;
  font-weight: 600;
  line-height: 31.200000000000003px;
  font-size: 12px;
  background-color: #ff7d3a;
  color: white;
}

.emotion-1:hover {
  background-color: #ff8e54;
  border: 1px solid #ff8e54;
  box-shadow: 4px 4px 15px rgba(255,125,58,0.1);
}

.emotion-1:focus {
  outline: none;
  background-color: #ff6c21;
  border: 1px solid #ed5000;
}

.emotion-1:active {
  background-color: #ff6c21;
  border: 1px solid #ff6c21;
  box-shadow: 'none';
}

.emotion-1:disabled {
  background-color: #f6f6f6;
  border: 1px solid #f3f3f3;
  color: #cdcdcd;
}

.emotion-1:disabled:hover {
  box-shadow: none;
}

.emotion-1 svg {
  margin: 0 8px 0 0;
  fill: white;
}

<Button
  disabled={false}
  iconDirection="left"
  mode="filled"
  size="m"
  stretched={false}
  type="warning"
>
  <Styled(withProps(Styled(button)))
    disabled={false}
    iconDirection="left"
    mode="filled"
    size="m"
    stretched={false}
    type="warning"
  >
    <withProps(Styled(button))
      className="emotion-3 emotion-0"
      disabled={false}
      iconDirection="left"
      mode="filled"
      size="m"
      stretched={false}
      type="warning"
    >
      <Styled(button)
        className="emotion-3 emotion-0"
        createProps={[Function]}
        disabled={false}
        iconDirection="left"
        mode="filled"
        size="m"
        stretched={false}
        type="warning"
      >
        <button
          className="emotion-0 emotion-1 emotion-2"
          disabled={false}
          mode="filled"
          size="m"
          type="warning"
        >
          Test
        </button>
      </Styled(button)>
    </withProps(Styled(button))>
  </Styled(withProps(Styled(button)))>
</Button>
`;
